packages/terra-collapse/tests/jest/Collapse.test.jsx

```diff
@@ -0,0 +1,134 @@
+  });
+
+  it('should set icon prop correctly', () => {
+    const collapse = shallow(<Collapse closedButtonText="Show">Test</Collapse>);
+    expect(collapse).toMatchSnapshot();
+  });
+
+  it('should set icon prop correctly with custom icon', () => {
+    const collapse = shallow(<Collapse closedButtonText="Show" icon={<IconCaretRight />}>Test</Collapse>);
+    expect(collapse).toMatchSnapshot();
+  });
+
+  it('should set isAnimated prop correctly', () => {
+    const collapse = shallow(<Collapse closedButtonText="Show" isAnimated>Test</Collapse>);
+    expect(collapse.unrendered.props.isAnimated).toEqual(true);
+  });
+
+  it('should set isInitiallyOpen prop correctly', () => {
+    const collapse = shallow(<Collapse closedButtonText="Show" isInitiallyOpen>Test</Collapse>);
+    expect(collapse.unrendered.props.isInitiallyOpen).toEqual(true);
+  });
+
+  it('should set onOpen prop correctly', () => {
+    const collapse = shallow(<Collapse closedButtonText="Show" onOpen={() => alert('Opened')}>Test</Collapse>);
+    expect(collapse.unrendered.props.onOpen.toString()).toEqual('function onOpen() {return alert(\'Opened\');}');
+  });
+
+  it('should set onClose prop correctly', () => {
+    const collapse = shallow(<Collapse closedButtonText="Show" onClose={() => alert('Closed')}>Test</Collapse>);
+    expect(collapse.unrendered.props.onClose.toString()).toEqual('function onClose() {return alert(\'Closed\');}');
+  });
+
+  it('should set openedButtonText prop correctly', () => {
+    const collapse = shallow(<Collapse closedButtonText="Show" openedButtonText="Hide">Test</Collapse>);
+    expect(collapse.unrendered.props.openedButtonText).toEqual('Hide');
+  });
+
+  // Structure Tests
+  it('should have the class terra-Collapse', () => {
+    const collapse = shallow(<Collapse closedButtonText="Show">Test</Collapse>);
+    expect(collapse.prop('className')).toContain('terra-Collapse');
+  });
+
+ // Attributes
+  it('should merge classes passed in with attributes', () => {
+    const collapse = shallow(<Collapse className="TestClass" closedButtonText="Show">Test</Collapse>);
+    expect(collapse.prop('className')).toContain('TestClass');
+  });
+
+  it('should merge ids passed in with attributes', () => {
+    const collapse = shallow(<Collapse id="TestId" closedButtonText="Show">Test</Collapse>);
+    expect(collapse.prop('id')).toContain('TestId');
+  });
+
+  it('should append data passed in with attributes', () => {
+    const collapse = shallow(<Collapse data-terra-text-mock="MockData" closedButtonText="Show">Test</Collapse>);
+    expect(collapse.prop('data-terra-text-mock')).toContain('MockData');
+  });
+
+  it('should append styles passed in with attributes', () => {
+    const collapse = shallow(<Collapse style={{ height: '100px' }} closedButtonText="Show">Test</Collapse>);
+    expect(collapse.prop('style')).toEqual({ height: '100px' });
+  });
+
+  // Error Handling Test
+  it('should throw error for required children', () => {
```

Comment:
Updated icon prop so its not required anymore with us providing a default prop for icon: https://github.com/cerner/terra-core/commit/817db96082dd8655ba479e573db950ab3acfed35